From 144894869a7284d21a7a0ab943dd50a46ddbd19c Mon Sep 17 00:00:00 2001
From: Josh Wilsdon <josh@wilsdon.ca>
Date: Thu, 22 Nov 2018 11:13:56 -0800
Subject: [PATCH] TOOLS-2114 sdcadm treats server as "unavailable" when it
 doesn't have a transitional_status Reviewed by: Trent Mick <trentm@gmail.com>
 Approved by: Trent Mick <trentm@gmail.com>

---
 CHANGES.md                        | 4 ++++
 lib/procedures/update-agent-v1.js | 3 ++-
 package.json                      | 2 +-
 3 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/CHANGES.md b/CHANGES.md
index 6cf671a..b269f56 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -10,6 +10,10 @@
 
 # sdcadm Changelog
 
+## 1.21.6
+
+- TOOLS-2114 sdcadm treats server as "unavailable" when it doesn't have a transitional_status
+
 ## 1.21.5
 
 - TOOLS-2110 sdcadm can crash when SAPI instance has undefined image
diff --git a/lib/procedures/update-agent-v1.js b/lib/procedures/update-agent-v1.js
index 95fda8f..c7c849f 100644
--- a/lib/procedures/update-agent-v1.js
+++ b/lib/procedures/update-agent-v1.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright 2017 Joyent, Inc.
+ * Copyright 2018 Joyent, Inc.
  */
 
 
@@ -255,6 +255,7 @@ UpdateAgentV1.prototype.execute = function uaExecute(opts, callback) {
                     servers.forEach(function (srv) {
                         if (srv.status !== 'running' ||
                             (srv.status === 'running' &&
+                             srv.hasOwnProperty('transitional_status') &&
                              srv.transitional_status !== '')) {
                             unavailable.push(srv.uuid);
                         }
diff --git a/package.json b/package.json
index fc73d37..3980180 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "sdcadm",
   "description": "Administer a SmartDataCenter (SDC) standup",
-  "version": "1.21.5",
+  "version": "1.21.6",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
-- 
2.21.0

